controller/contacts.js

```diff
@@ -0,0 +1,127 @@
+const service = require("../service");
+const { patternContact, patternFavorite } = require("../joi");
+
+const get = async (req, res, next) => {
+  try {
+    const contacts = await service.getAllContacts();
+    res.json({ data: contacts, status: "success", code: 200 });
+  } catch (err) {
+    next(err);
+  }
+};
+
+const getById = async (req, res, next) => {
+  const { contactId } = req.params;
+  try {
+    const contact = await service.getContactById(contactId);
+    res.json({ data: contact, status: "success", code: 200 });
+  } catch (err) {
+    next(res.json({ message: "Not found", status: "error", code: 404 }));
+  }
+};
+
+const add = async (req, res, next) => {
+  const validated = patternContact.validate(req.body);
+  if (validated.error) {
+    res.json({ message: validated.error.message, status: "error", code: 400 });
+  }
+  const { name, email, phone } = validated.value;
+
+  if (name && email && phone) {
```

Comment:
Czy jest potrzeba tutaj strzelać ifa na te wartości skoro walidujesz je w 24 linijce?

Reply:
W zadaniu był opisany konkretny message jaki powinien byc zwrócony gdyby brakowało któregoś z pól. Naturalnie normalnie wystarczyłoby dac required() w Joi jednak w tym wypadku musiałem to pominąć i z ifować w controllerze.

Reply:
Tutaj rzeczywiście nie było potrzeby ustawiać ifa bo required() w Joi i res.json() w opracowaniu errora walidacji wystarczy. Poprawione :)